frontend/src/features/entity_selector/AutoEntityButton.tsx

```diff
@@ -20,22 +21,43 @@ const AutoEntityButton = ({
   onResponse,
   onClick,
   disabled = false,
+  autoFill = false,
 }: Props) => {
   const { t } = useTranslation();
   const { name: pollName } = useCurrentPoll();
+  const mountedRef = useRef(false);
 
-  const askNewVideo = async () => {
+  const askNewVideo = useCallback(async () => {
     onClick();
     const newVideoId: string | null = await getVideoForComparison(
       otherUid ? idFromUid(otherUid) : null,
-      idFromUid(currentUid)
+      idFromUid(currentUid || '')
     );
+
+    if (!mountedRef.current) return;
+
     if (newVideoId) {
       onResponse(`${UID_YT_NAMESPACE}${newVideoId}`);
     } else {
       onResponse(null);
     }
-  };
+  }, [currentUid, otherUid, onClick, onResponse]);
+
+  useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  const previousUidRef = useRef<string | null | undefined>(undefined);
+  useEffect(() => {
+    // We only want to autofill when the currentUid changes, not when the other dependencies change
+    if (currentUid === previousUidRef.current) return;
+    previousUidRef.current = currentUid;
+
+    if (autoFill && currentUid === null) askNewVideo();
```

Comment:
The only problem I found is that it may place twice the same video:
<img width="894" alt="Screenshot 2023-01-30 at 23 03 32" src="https://user-images.githubusercontent.com/8818081/215605989-deb97f28-ae94-4add-abe1-037ded822457.png">

It occurs particularly often with a single (or few) video in the rate later list (this maybe be often the case for users)

Could it be improved with the following behaviour: The component should keep track that the incoming selected uid comes from the autofill mechanism. If it happens to be equal to the changed property "other_id" then autofill is re-triggered?

Reply:
Yes we can probably do that by checking the updated `otherUid` after the `newVideoId` is received (only when `askNewVideo` was called from the auto fill). We will probably need another ref for that because `askNewVideo` can't access the props updated after it was called.

I think it would still be possible to have identical ids loaded when both responses arrives at the exact same time and the `otherUid` has not yet been propagated by a new render. But I think it should be rare or even impossible (it depends on the internals of React).

Another solution could be to move this autofill mechanism up to a component having both uids and having a special process to request the ids when both are null.

I'll try the `otherUid` method.

Reply:
Done in 89b60737589832c92c2f89621f556f6a2d1d6bf2.